src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -440,7 +442,12 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return str_replace('\\', '_', \get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
+        $name = \get_class($this);
+        if (false !== $i = strrpos($name, '\\')) {
+            $name = substr($name, 1 + $i);
```

Comment:
keeping only the last part is enough IMHO

Reply:
I would say quite the contrary for multi-app projects. Let's stay on the safe side and keep the FQDN.

Reply:
ok, reverted